tests/0170-share_consumer_subscription.c

```diff
@@ -0,0 +1,530 @@
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "test.h"
+
+/*
+ * Share consumer subscription tests.
+ *
+ * Tests subscription/unsubscription mechanics including:
+ * - Subscribed topics via rd_kafka_subscription()
+ * - Unsubscribe idempotence
+ * - Resubscription behavior (topic replacement)
+ * - Subscribing to non-existent topics
+ * - Polling without active subscription
+ */
+
+#define BATCH_SIZE 128
+
+
+/**
+ * Create a share consumer with standard configuration.
+ */
+static rd_kafka_t *create_share_consumer(const char *group) {
+        char errstr[512];
+        rd_kafka_conf_t *conf;
+
+        test_conf_init(&conf, NULL, 60);
+        rd_kafka_conf_set(conf, "share.consumer", "true", errstr, sizeof(errstr));
+        rd_kafka_conf_set(conf, "group.protocol", "consumer", errstr, sizeof(errstr));
+        rd_kafka_conf_set(conf, "group.id", group, errstr, sizeof(errstr));
+        rd_kafka_conf_set(conf, "enable.auto.commit", "false", errstr, sizeof(errstr));
+
+        rd_kafka_t *rk = rd_kafka_new(RD_KAFKA_CONSUMER, conf, errstr, sizeof(errstr));
+        TEST_ASSERT(rk, "Failed to create share consumer: %s", errstr);
+        return rk;
+}
+
+/**
+ * Subscribe to a list of topics.
+ */
+static void subscribe_topics(rd_kafka_t *rk, const char **topics, int cnt) {
+        rd_kafka_topic_partition_list_t *subs;
+        int i;
+
+        subs = rd_kafka_topic_partition_list_new(cnt);
+        for (i = 0; i < cnt; i++)
+                rd_kafka_topic_partition_list_add(subs, topics[i], RD_KAFKA_PARTITION_UA);
+
+        TEST_CALL_ERR__(rd_kafka_subscribe(rk, subs));
+        rd_kafka_topic_partition_list_destroy(subs);
+}
```

Comment:
you can use `test_consumer_subscribe_multi` instead. If this doesn't suffice, I have made a generic multi subscribe method in on of my opened [PR](https://github.com/confluentinc/librdkafka/pull/5203/changes#diff-cebecad5a3105afd7d32b8dd7e2ea09203a488c26f8f43ac40914cf1cf7e31dfL595). I will try to get that merged before this if that is required.